pkg/cloud/gcp/actuators/machineset/cache.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machineset
+
+import (
+	"fmt"
+	"sync"
+
+	computeservice "github.com/openshift/cluster-api-provider-gcp/pkg/cloud/gcp/actuators/services/compute"
+	gce "google.golang.org/api/compute/v1"
+)
+
+// MachineTypeKey is used to identify MachineType.
+type MachineTypeKey struct {
+	Zone        string
+	MachineType string
+}
+
+// machineCache is used for caching machine types.
+type machineCache struct {
+	cacheMutex    sync.Mutex
+	machinesCache map[MachineTypeKey]*gce.MachineType
+}
+
+// newMachineCache creates empty machineCache.
+func newMachineCache() *machineCache {
+	return &machineCache{
+		machinesCache: map[MachineTypeKey]*gce.MachineType{},
+	}
+}
+
+// GetMachineFromCache retrieves machine type from cache under lock.
+func (mc *machineCache) getMachineFromCache(gcpService computeservice.GCPComputeService, projectID string, zone string, machineType string) (*gce.MachineType, error) {
+	mc.cacheMutex.Lock()
+	defer mc.cacheMutex.Unlock()
+
+	// Machine Type already fetched from GCE
+	if machine, ok := mc.machinesCache[MachineTypeKey{zone, machineType}]; ok {
```

Comment:
s/machine/machineType in 49, 53 and 59?

Reply:
One of the function args is `machineType` so I renamed it to `mt`, could also rename the function argument to something like `machineTypeName` but it's plumbed through from a field that is `MachineType`